modules/lib/src/main/resources/assets/js/app/inputtype/ui/text/dialog/MacroModalDialog.ts

```diff
@@ -202,8 +202,8 @@ export class MacroModalDialog
     }
 
     private insertUpdatedMacroIntoTextArea(macroString: string) {
-        this.selectedMacro.element.setText(
-            this.selectedMacro.element.getText().replace(this.selectedMacro.macroText, macroString));
+        this.selectedMacro.element.$.innerText =
+            this.selectedMacro.element.$.innerText.replace(this.selectedMacro.macroText, macroString);
```

Comment:
@ashklianko Is it `setText` or `getText` that does sanitisation in CKE? Can we keep `setText` and only replace `getText` with `$.innerText`?

Reply:
setText and getText don't preserve line breaks, so just innerText works